common/src/IO/ObjSerializer.cpp

```diff
@@ -107,7 +107,7 @@ namespace TrenchBroom {
             for (const auto& [textureName, texture] : usedTextures) {
                 str << "newmtl " << textureName << "\n";
                 if (texture != nullptr && !texture->relativePath().isEmpty()) {
-                    str << "map_Kd " << texture->relativePath().asString() << "\n\n";
+                    str << "map_Kd " << mtlPath.deleteLastComponent().makeRelative(texture->absolutePath()).asString() << "\n\n";
```

Comment:
I would extract `mtlPath.deleteLastComponent()` as a variable before the loop.

Reply:
Also, `Path` now has an `operator<<` overload, so you can drop the `.asString()` while you're here.